docs/Roles and Access Control/3.API Keys.md

```diff
@@ -33,3 +35,39 @@ To create and configure access control for API keys:
     ```
 
     > In [settings](https://github.com/epam/ai-dial-core?tab=readme-ov-file#dynamic-settings), you can get familiar with the description of the configuration parameters. Refer to [configuration file](https://github.com/epam/ai-dial-core/blob/development/sample/aidial.config.json) to view the full example. 
+
+## Per-Request Keys
+
+Per-request keys are used to manage access to user files for applications, enable open telemetry for tracing and realize cost control in a lifespan of a particular request.
+
+Per-request keys are generated by AI DIAL Core, when it is making a request to the application and is valid only during the lifetime of this particular request from the Core to the application.
+
+**Flow**:
+
+1. AI DIAL Core checks that the caller (chat user or application) has enough permissions and limits (based on the [configuration](#api-keys)) to perform the request and generates a new per-request key specifically for this request.
+2. AI DIAL Core stores an additional data associated with the per-request key either in memory or in Redis. The data may include but is not limited to: JWT of the user or API key of the application, `trace-id`, `core-parent-span-id`, list of the user files attached to the request.
+3. AI DIAL Core passes the per-request key to the application or model adapter in the API-key header. The application or model adapter then send the per-request key back to AI DIAL Core using the same API-key header for model requests or file storage API requests.
+4. AI DIAL Core invalidates the per-request key when the request is complete.
+
+### Calculating Costs & Limits
+
+Per-request keys can be used to attribute costs and limits incurred by a request to a model to the user or application that initiated the request.
+
+To attribute costs and limits to the request originator, `trace-id` should be included in the request header. Incorrect `trace-id` will fail the request with the error 401 unauthorized. `trace-id` is saved throughout the lifespan of the request in memory or in Redis.
```

Comment:
change `To attribute costs and limits to the request originator, trace-id should be included in the request header.` -> `To attribute costs and limits to the request originator, `trace parent` should be included in the request header.`

Reply:
changed